### ISSUE TITLE
The first step of TPI

### ISSUE THREAD

**Issue by JLian401** [jdebacker/ProblemSet9#27]
From chapter 8.6, we know that we should guess BQ in the first step of TPI. Currently, we have a get_BQ function in aggregate.py, where we input r and other parameters to return BQ. But I think that is not we can use in the first step of TPI and that we should define it in the SS function. Is it the right way?

**Comment by jdebacker** [comment]
@JLian401 Yes, at least for the time path solution, you will need to guess the path of BQ.  You can start with an initial guess of, e.g., BQ_path = np.ones(T) * BQ_ss, but then with each iteration through the time path you will use the `get_BQ` function to update this guess, just as you did with the `r_path`.

**Comment by saharnazb** [comment]
I started like this: (But not sure if I am doing right with (T + S)

![image](https://user-images.githubusercontent.com/53350566/70679443-24070000-1c63-11ea-91b8-bc345cebe87a.png)

**Comment by jdebacker** [comment]
@saharnazb That look like a good starts.   But you'll want both the `r_path` and `BQ_path` to be the same length, `T+S`.

**Comment by JLian401** [comment]
@saharnazb In my understanding, maybe we can just set BQ_init as one input then make BQ_path = np.ones(T+S) * BQ_ss. Then we will no need to define BQ_path_init elsewhere and no need to use np.append .

**Comment by saharnazb** [comment]
@JLian401 That can make it simple. let's make sure it does not make any conflict with anything else.

**Comment by saharnazb** [comment]
@JLian401 I am thinking that we should keep `_path_init` because using `_init` the convergence may take longer but `_path_init` is defined based on `r_ss`. To get `r_path`, starting from `_path_init` will make the convergence faster

**Comment by saharnazb** [comment]
@jdebacker I am trying to use `np.tri()` for the time path solution. Is this the right way to solve? I was thinking to have zeros at and below the first full lifetime decision-maker. Or should I have the partial decision-makers' path separately?

**Comment by jdebacker** [comment]
@saharnazb You won't need to use `np.tri()`.  Look a the [3 period solution ](https://github.com/jdebacker/CompEcon_Fall19/blob/master/OverlappingGenerations/3PeriodModel/TPI.py), use a similar "masking" matrix...

**Comment by saharnazb** [comment]
@jdebacker I mean for the ones that are not full lifetime. I already defined them using `np.tri()` and iterated on `t`. So I should change it. But it seems someone else has done it in a different way. Should I continue?
